tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -1143,7 +1144,32 @@ static unique_ptr<TableRef> TryReplacement(py::dict &dict, py::str &table_name,
 		auto materialized = entry.attr("collect")();
 		auto arrow_dataset = materialized.attr("to_arrow")();
 		CreateArrowScan(arrow_dataset, *table_function, children, config);
-	} else {
+	} else if ((numpytype = DuckDBPyConnection::IsAcceptedNumpyObject(entry)) > 0) {
+                string name = "np_" + GenerateRandomName();
+                py::dict data; // we will convert all the supported format to dict{"key": np.array(value)}.
+                if (numpytype == 1) {
+                        // numpy array with shape (n, )
+                        data["column0"] = entry;
+                } else if (numpytype == 2 || numpytype == 3) {
+                        // numpy array with shape (n_rows, n_cols) or list of numpy arrays of shape (n,)
+                        size_t idx = 0;
+                        for(auto item : py::array(entry)) {
+                                data[("column" + std::to_string(idx)).c_str()] = item;
+                                idx++;
+                        }
+                } else if (numpytype == 4) {
+			// {"key": np.array}
+                        data = py::cast<py::dict>(entry);
+                } else {
+                        throw InvalidInputException("Invalid type of known numpy obj");
+                }
+                auto new_df = py::module_::import("pandas").attr("DataFrame").attr("from_dict")(data);
```

Comment:
Please use the import_cache where available

If I'm not mistaken, this should work:
```c++
import_cache.pandas().DataFrame().attr("from_dict")(data);
```

Reply:
Also, I'm not sure if this is what we want to do here

The idea is to be able to pass numpy arrays without necessarily having to depend on `pandas`, no?
On top of that, pandas DataFrames created from numpy arrays aren't guaranteed to keep their original type, so I'd prefer it if we could create a way to transform a dictionary/list of numpy arrays directly to a DataChunk, instead of first going to a pandas DataFrame.